// @smash-sdk/transfer/10-2019/types/LockTransfer/LockTransferError.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GatewayTimeoutError = exports.BadGatewayError = exports.InternalServerError = exports.NotFoundError = exports.TransferIsInQueueError = exports.TransferIsDirtyError = exports.TransferAlreadyLockedError = exports.UnauthorizedError = exports.BadRequestError = void 0;
const core_1 = require("@smash-sdk/core");
class BadRequestError extends core_1.SDKError {
}
exports.BadRequestError = BadRequestError;
class UnauthorizedError extends core_1.SDKError {
}
exports.UnauthorizedError = UnauthorizedError;
class TransferAlreadyLockedError extends core_1.SDKError {
}
exports.TransferAlreadyLockedError = TransferAlreadyLockedError;
class TransferIsDirtyError extends core_1.SDKError {
}
exports.TransferIsDirtyError = TransferIsDirtyError;
class TransferIsInQueueError extends core_1.SDKError {
}
exports.TransferIsInQueueError = TransferIsInQueueError;
class NotFoundError extends core_1.SDKError {
}
exports.NotFoundError = NotFoundError;
class InternalServerError extends core_1.SDKError {
}
exports.InternalServerError = InternalServerError;
class BadGatewayError extends core_1.SDKError {
}
exports.BadGatewayError = BadGatewayError;
class GatewayTimeoutError extends core_1.SDKError {
}
exports.GatewayTimeoutError = GatewayTimeoutError;
exports.default = {
    BadRequestError,
    UnauthorizedError,
    TransferAlreadyLockedError,
    TransferIsDirtyError,
    TransferIsInQueueError,
    NotFoundError,
    InternalServerError,
    BadGatewayError,
    GatewayTimeoutError,
};
