// @smash-sdk/transfer/07-2022/types/DeleteUserTransfer/DeleteUserTransferError.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GatewayTimeoutError = exports.BadGatewayError = exports.InternalServerError = exports.NotFoundError = exports.TransferNotUploadedError = exports.UnauthorizedError = exports.BadRequestError = void 0;
const core_1 = require("@smash-sdk/core");
class BadRequestError extends core_1.SDKError {
}
exports.BadRequestError = BadRequestError;
class UnauthorizedError extends core_1.SDKError {
}
exports.UnauthorizedError = UnauthorizedError;
class TransferNotUploadedError extends core_1.SDKError {
}
exports.TransferNotUploadedError = TransferNotUploadedError;
class NotFoundError extends core_1.SDKError {
}
exports.NotFoundError = NotFoundError;
class InternalServerError extends core_1.SDKError {
}
exports.InternalServerError = InternalServerError;
class BadGatewayError extends core_1.SDKError {
}
exports.BadGatewayError = BadGatewayError;
class GatewayTimeoutError extends core_1.SDKError {
}
exports.GatewayTimeoutError = GatewayTimeoutError;
exports.default = {
    BadRequestError,
    UnauthorizedError,
    TransferNotUploadedError,
    NotFoundError,
    InternalServerError,
    BadGatewayError,
    GatewayTimeoutError,
};
