// @smash-sdk/transfer/07-2020/types/CreateTransferFile/CreateTransferFileError.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GatewayTimeoutError = exports.BadGatewayError = exports.InternalServerError = exports.ConflictError = exports.NotFoundError = exports.TransferIsInQueueError = exports.TransferAlreadyLockedError = exports.UnauthorizedError = exports.BadRequestError = void 0;
const core_1 = require("@smash-sdk/core");
class BadRequestError extends core_1.SDKError {
}
exports.BadRequestError = BadRequestError;
class UnauthorizedError extends core_1.SDKError {
}
exports.UnauthorizedError = UnauthorizedError;
class TransferAlreadyLockedError extends core_1.SDKError {
}
exports.TransferAlreadyLockedError = TransferAlreadyLockedError;
class TransferIsInQueueError extends core_1.SDKError {
}
exports.TransferIsInQueueError = TransferIsInQueueError;
class NotFoundError extends core_1.SDKError {
}
exports.NotFoundError = NotFoundError;
class ConflictError extends core_1.SDKError {
}
exports.ConflictError = ConflictError;
class InternalServerError extends core_1.SDKError {
}
exports.InternalServerError = InternalServerError;
class BadGatewayError extends core_1.SDKError {
}
exports.BadGatewayError = BadGatewayError;
class GatewayTimeoutError extends core_1.SDKError {
}
exports.GatewayTimeoutError = GatewayTimeoutError;
exports.default = {
    BadRequestError,
    UnauthorizedError,
    TransferAlreadyLockedError,
    TransferIsInQueueError,
    NotFoundError,
    ConflictError,
    InternalServerError,
    BadGatewayError,
    GatewayTimeoutError,
};
