// @smash-sdk/core/dist/errors/sdkError.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkError = exports.UnknownError = exports.SDKError = void 0;
class SDKError extends Error {
    constructor(error) {
        super();
        if (typeof error === 'string') {
            this.name = "SDKError";
            this.message = error;
        }
        else if (error instanceof Error) {
            this.name = error.name;
            this.message = error.message;
            this.stack = error.stack;
        }
        else if (error.name) {
            this.message = error.error;
            Object.assign(this, error);
            this.stack = SDKError.stringifiedStack();
        }
        else {
            this.name = 'Unknown SDK error';
            this.message = 'Unknown SDK error';
        }
    }
    static stack() {
        const prepareStackTrace = Error.prepareStackTrace;
        Error.prepareStackTrace = (_, stack) => stack;
        const stack = (new Error().stack).filter(s => !!s.getFileName());
        while (stack[0].getFileName().indexOf("sdkError") >= 0) {
            stack.shift();
        }
        while (stack[0].getFileName().indexOf("@smash-sdk") >= 0) {
            stack.shift();
        }
        Error.prepareStackTrace = prepareStackTrace;
        return stack;
    }
    static stringifiedStack() {
        return SDKError.stack().join("\n");
    }
}
exports.SDKError = SDKError;
class UnknownError extends SDKError {
}
exports.UnknownError = UnknownError;
class NetworkError extends SDKError {
}
exports.NetworkError = NetworkError;
